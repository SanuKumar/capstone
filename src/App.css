body {
  background-image: url(https://img.freepik.com/free-vector/hand-painted-watercolor-pastel-sky-background_23-2148902771.jpg?w=2000);
  background-position: center;
  background-size: cover;
  /* background-repeat: no-repeat; */
}

.card-text-formatter {
  height: 2em;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.footer {
  position: fixed;
  padding: 10px 10px 0px 10px;
  bottom: 0;
  width: 100%;
  /* Height of the footer*/
  height: 40px;
  background: grey;
}

.loader {
  margin: 10rem 0 0 50%;
}

.form-error-msg {
  color: red;
}
