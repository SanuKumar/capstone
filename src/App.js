import React, { useEffect, useState } from "react"
import './App.css';
import Header from "./components/Header";
import Footer from "./components/Footer";
import axios from "axios";
import Product from "./pages/Product"
import {
  Switch,
  Route,
} from "react-router-dom";
import Login from "./pages/Login";
import Register from "./pages/Register";
import About from "./pages/About";
import ChartPage from "./pages/ChartPage";
import AddProduct from "./pages/AddProduct";
import PageNotFound from "./pages/PageNotFound.js"
import { Suspense, lazy } from 'react';
import Loader from "./components/Loader";

const ProductList = lazy(() => import('./components/ProductList'));

const App = () => {
  const [products, setProducts] = useState([]);
  const [searchResult, setSearchResult] = useState([])
  const [isUserLoggedIn, setIsUserLoggedIn] = useState(null);
  const [loading, setLoading] = useState(false)

  useEffect(() => {
    fetchProduct();
  }, [])

  const fetchProduct = async () => {
    setLoading(true)
    try {
      const res = await axios.get(`http://localhost:3002/products`)
      setLoading(false)
      let reversedData = res.data.reverse()
      setProducts(reversedData)
      setSearchResult(reversedData)
    } catch (error) {
      console.log(error)
      setLoading(false)
    }
  }

  useEffect(() => {
    setIsUserLoggedIn(JSON.parse(localStorage.getItem('isUserLoggedIn')))
  }, [])

  const fetchProductCallBack = () => {
    fetchProduct()
  }

  const handleProductSearch = (searchKey) => {
    if (searchKey) {
      let filterProduct = searchResult.filter((product) => product.title.toLowerCase().includes(searchKey.toLowerCase()));
      setProducts(filterProduct)
    } else {
      setProducts(searchResult)
    }
  }

  return (
    <>
      <Header handleProductSearch={handleProductSearch} isUserLoggedIn={isUserLoggedIn} />
      <Suspense fallback={<Loader />}>
        <div style={{ padding: "10rem 0 5rem 0" }}>
          <Switch>
            <Route exact path="/" ><ProductList products={products} loading={loading} isUserLoggedIn={isUserLoggedIn} fetchProductCallBack={fetchProductCallBack} /></Route>
            <Route exact path="/product/:id"><Product isUserLoggedIn={isUserLoggedIn} fetchProductCallBack={fetchProductCallBack} /></Route>
            <Route exact path="/login"><Login /></Route>
            <Route exact path="/register"><Register /></Route>
            <Route exact path="/about"><About /></Route>
            <Route exact path="/chart"><ChartPage products={products} /></Route>
            <Route exact path="/addproduct"><AddProduct fetchProductCallBack={fetchProductCallBack} /></Route>
            <Route path="*" component={PageNotFound} />
          </Switch>
        </div>
      </Suspense>
      <Footer />
    </>
  );
}

export default App;
