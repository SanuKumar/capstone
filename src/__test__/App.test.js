import React from "react"
import { render, cleanup } from "@testing-library/react"
import App from "../App"

afterEach(cleanup)

describe("App page snapshot", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  })
})