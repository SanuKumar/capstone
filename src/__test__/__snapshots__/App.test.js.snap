// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
        style="cursor: pointer;"
      >
        Product Inventory
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            href="#"
            role="button"
            tabindex="0"
          >
            About
          </a>
          <a
            class="nav-link"
            href="#"
            role="button"
            style="margin-right: 2rem;"
            tabindex="0"
          >
            Product's Chart
          </a>
          <form
            class="d-flex"
          >
            <input
              aria-label="Search"
              class="me-2 form-control"
              placeholder="Search"
              type="search"
            />
          </form>
        </div>
      </div>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <div
          class="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            class="dropdown-toggle nav-link"
            href="#"
            id="basic-nav-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              style="color: white;"
            >
              Login/Register
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="loader"
  >
    <div
      class="spinner-border spinner-border-xl"
    />
  </div>
  <footer
    class="footer"
  >
    <p
      style="color: white; text-align: center;"
    >
      Sanu Â©. All rights reserved.
    </p>
  </footer>
</DocumentFragment>
`;
